Sources/FirebaseFirestore/DocumentReference+Swift.swift

```diff
@@ -64,6 +64,35 @@ extension DocumentReference {
 
     return ListenerRegistration(boxed, instance)
   }
+
+  public func setData(_ data: [String: Any]) async throws -> Void {
```

Comment:
```suggestion
  private func setData(_ data: [String: Any]) async throws -> Void {
```

https://firebase.google.com/docs/reference/swift/firebasefirestore/api/reference/Classes/DocumentReference#setdata_:

Seems to indicate that the public surface is synchronous.

Reply:
We seem to have async/await versions directly available in our code https://github.com/thebrowsercompany/arc/blob/0e80a115cdad70b66a68750817aa470206e0d29a/Frameworks/ARCClients/Sources/BoostsClient/BoostsStorageFirebaseImpl.swift#L223 is this some automatic conversion from callback to async/await that Apple does?

Reply:
Yes, this could be the a remapping that Apple does.  I think that we should verify that in the headers and match that.  It is also possible that they have a Swift extension that they are shipping?  I'm just lost how that is getting mismatched.  If they have variants through an overlay, we should consider pulling these out into an extension of the type and map the core type to the core type.

Reply:
Interesting... clicking on the headers from that same call site I linked to just takes me to https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Source/Public/FirebaseFirestore/FIRDocumentReference.h#L144 so I guess Apple is doing some kind of conversion. Let me see if that will magically work for us on Windows...

Reply:
So I decided to keep the public async interfaces since we are using them, but to also expose a block based API that the objective-c files seem to have as their primary interface. In order to keep the future's alive for long enough to do their job but to not change the function signature I wait for them to resolve as an item in a dispatch queue. This sees to work, but I wonder if there could be some issues here....

Reply:
LMK if you want to pair on this today if you have a different idea in mind!